server/aws-lsp-codewhisperer/src/language-server/agenticChat/context/contextCommandsProvider.ts

```diff
@@ -0,0 +1,147 @@
+import * as path from 'path'
+import { FSWatcher, watch } from 'chokidar'
+import { Features } from '../../types'
+import { ContextCommand, ContextCommandGroup } from '@aws/language-server-runtimes/protocol'
+import { TriggerContext } from '../../chat/contexts/triggerContext'
+import { URI } from 'vscode-uri'
+import { getUserHomeDir } from '@aws/lsp-core/out/util/path'
+
+export type AdditionalContextLengths = {
+    fileContextLength: number
+    promptContextLength: number
+    ruleContextLength: number
+}
+
+export class ContextCommandsProvider {
```

Comment:
does this codebase need a `Disposable` interface?
```suggestion
export class ContextCommandsProvider implements Disposable {
```

Reply:
I guess we can use the one from `vscode-jsonrpc`, will do